### PR TITLE
lua: test the interpreter

### PR DESCRIPTION
**Pull request by teto**
add some basic tests to the lua interpreter. Some aspects of the nix lua ecosystem are certainly broken so I would like to have some tests to improve it and prevent regressions.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: lib/tests/assert.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Always failing assertion with a message.
```
Comment (roberth) [inline lib/tests/assert.sh:1]: `lib/tests` is for code that supports the testing of `lib` itself. Please move this file near where it's used.

File: pkgs/test/make-wrapper/lib.nix [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (roberth) [inline pkgs/test/make-wrapper/lib.nix:1]: Please keep it near your test for this PR. You can follow it up with a proposal to make it available to other code later, which is a different goal from testing lua, and needs different reviewers. By splitting the work, your PR will be easier to merge.

File: pkgs/development/interpreters/lua-5/tests.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+let
+
+  wrapLib = import ../../../test/make-wrapper/lib.nix;
+
+  nmt = fetchFromGitLab {
+    owner = "rycee";
+    repo = "nmt";
+    rev = "d2cc8c1042b1c2511f68f40e2790a8c0e29eeb42";
+    sha256 = "1ykcvyx82nhdq167kbnpgwkgjib8ii7c92y3427v986n2s5lsskc";
+  };
```
Comment (roberth) [inline pkgs/development/interpreters/lua-5/tests.nix:18]: `nmt` here is unused code.

**Review by roberth — COMMENTED** [review]
I appreciate your intent to allow your code to be reused, but this is a separate task that will both delay your PR and make your reusable test framework idea less successful because it deserves attention from a different audience of reviewers and collaborators, who may have little interest in lua.

**Comment by figsoda** [comment]
does it rely on the changes in staging? if not, perhaps we should rebase it to master

**Pull request closed without merging**